src/func/collection.rs

```diff
@@ -619,3 +636,23 @@ impl SearchResult {
         format!("{:?}", self)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn config_with_distance() {
+        for dist in vec!["cosine", "dot", "euclidean"] {
+            let conf = Config::new(10, 10, 3.0, dist).unwrap();
+        }
+    }
+
+    #[test]
+    #[should_panic(expected = "Distance not supported")]
+    fn config_with_distance_panic() {
+        for dist in vec!["l2", "test"] {
+            let conf = Config::new(10, 10, 3.0, dist).unwrap();
+        }
+    }
+}
```

Comment:
Can we move the test to `tests/test_collection`? We want to keep this file clean and dedicated for the functionality.

Reply:
Sure, I will fix that.